src/components/update-service/update-service.controller.ts

```diff
@@ -0,0 +1,348 @@
+      this.configStep.valid = isValid && this.configChanged;
+      this.reviewStep.allowed = this.configStep.valid;
+    });
+
+    if (this.dataWatcher) {
+      this.dataWatcher();
+    }
+
+    this.dataWatcher = this.$scope.$watch(() => {
+      return this.ctrl.parameterData;
+    }, () => {
+      this.configChanged = !angular.equals(this.getParameters(this.ctrl.parameterData), this.origParameterData) || (this.ctrl.selectedPlan !== this.originalPlan);
+      this.configStep.valid = _.get(this.ctrl, 'forms.orderConfigureForm.$valid') && this.configChanged;
+    }, true);
+
+    this.ctrl.nextTitle = 'Update';
+  };
+
+  public showResults = () => {
+    this.clearValidityWatcher();
+    this.ctrl.configPageShown = false;
+    this.ctrl.nextTitle = "Close";
+    this.ctrl.wizardDone = true;
+    this.updateServiceConfig();
+  };
+
+  public selectPlan = (plan: any) => {
+    this.ctrl.selectedPlan = plan;
+
+    if (_.get(plan, 'metadata.name') === _.get(this.ctrl.serviceInstance, 'spec.clusterServicePlanRef.name')) {
+      this.ctrl.parameterData = angular.copy(this.origParameterData);
+    } else {
+      // Clear any previous parameter data since each plan has its own parameter schema.
+      this.ctrl.parameterData = {};
+    }
+    this.updateParameterSchema(plan);
+
+    this.configChanged = !angular.equals(this.ctrl.parameterData, this.origParameterData) || (this.ctrl.selectedPlan !== this.originalPlan);
+    this.configStep.valid = _.get(this.ctrl, 'forms.orderConfigureForm.$valid') && this.configChanged;
+  };
+
+  public updateServiceConfig = () => {
+    this.ctrl.orderComplete = false;
+    this.ctrl.error = null;
+
+    var parameterData: any = this.getParameters(this.ctrl.parameterData);
+
+    // Update the instance parameters
+    var originalParameters: any = _.get(this.ctrl.serviceInstance, 'spec.parameters');
+    var updateParameters: any = _.pick(parameterData, _.keys(originalParameters));
+    var newParameters: any = _.omit(parameterData, _.keys(originalParameters));
+    var updateInstance: any = angular.copy(this.ctrl.serviceInstance);
+
+    if (_.get(updateInstance, 'spec.externalClusterServicePlanName') !== _.get(this.ctrl.selectedPlan, 'spec.externalName')) {
+      _.set(updateInstance, 'spec.clusterServicePlanRef', undefined);
+      _.set(updateInstance, 'spec.externalClusterServicePlanName', _.get(this.ctrl.selectedPlan, 'spec.externalName'));
+    }
+
+    if (!angular.equals(updateParameters, originalParameters)) {
+      _.set(updateInstance, 'spec.parameters', updateParameters);
+    }
+
+    // Update the secret parameters
+    var updatedSecretParameters: any = {};
+
+    _.each(this.secrets, (secret: any) => {
+      var originalSecretParameters: any = JSON.parse(this.SecretsService.decodeSecretData(secret.data).parameters);
+      var updateSecretParameters: any = _.pick(parameterData, _.keys(originalSecretParameters));
+      newParameters = _.omit(newParameters, _.keys(originalSecretParameters));
```

Comment:
I don't think this handles property names with a `.` in them correctly. It will treat them like JSON object paths.

Maybe (untested)

```js
var keys = _.map(originalSecretParameters, (value: any, key: string) => {
  // Make sure properties with a `.` in the name aren't treated as paths.
  return [ key ];
};
var updateSecretParameters: any = _.pick(parameterData, keys);
newParameters = _.omit(newParameters, keys);
```

Reply:
Tested and this is working as is when there is a . in the key. I can update if you aren't comfortable though.

Reply:
I went with your suggestion, cleaner to get the keys just once anyhow.